roof_classify/logic/classification.py

```diff
@@ -150,6 +151,20 @@ def labellingRoofingRaster(
             labelledImg += roofLabelledRaster
         return labelledImg
 
+    @staticmethod
+    def write_link_classlabel_roof_shp(roofingShapefileDir):
+        shpFolder = Path(roofingShapefileDir).rglob("*.shp")
+        files = [x for x in shpFolder]
+        # Write into a csv file the link between class label and roof layer
+        csv_filepath = roofingShapefileDir + "/link_classlabel_rooflayer.csv"
+        f = open(csv_filepath, "w")
+        writer = csv.writer(f)
+        fields = ["class label", "roof layer filepath"]
+        writer.writerow(fields)
+        for classLabel, roofVectorFilepath in enumerate(files, start=1):
+            writer.writerow([classLabel, roofVectorFilepath])
+        f.close()
```

Comment:
Use context manager and pthlib to improve code reliability:

```suggestion
    def write_link_classlabel_roof_shp(roofingShapefileDir: str):
        shpFolder = Path(roofingShapefileDir).rglob("*.shp")

        # Write into a csv file the link between class label and roof layer
        csv_filepath = Path(roofingShapefileDir, "link_classlabel_rooflayer.csv")
        with csv_filepath.open(mode="w", encoding="UTF8") as f:
            writer = csv.writer(f)
            fields = ["class label", "roof layer filepath"]
            writer.writerow(fields)
            for classLabel, roofVectorFilepath in enumerate(files, start=1):
                writer.writerow([classLabel, roofVectorFilepath])
```

Furthermore, note sure that writing/reading a CSV is not a bit overkill to save keys/values.

Reply:
Please add docstrin @quythytruong

Reply:
Solved by b0e2b4df0c34d5335d827bb356c7e959ee8dd559